src/device.rs

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use actix::{prelude::*, Actor, Context};
+use log::{debug, error, trace};
+use serde_json::Value;
+use std::collections::HashMap;
+use webthings_gateway_ipc_types::{Device as DeviceDescription, Property as PropertyDescription};
+
+pub struct Device {
+    description: DeviceDescription,
+}
+
+impl Device {
+    pub fn new(description: DeviceDescription) -> Self {
+        Self { description }
+    }
+}
+
+impl Actor for Device {
+    type Context = Context<Self>;
+
+    fn started(&mut self, _ctx: &mut Context<Self>) {
+        info!("Device started {}", self.description.id);
+    }
+
+    fn stopped(&mut self, _ctx: &mut Context<Self>) {
+        info!("Device stopped {}", self.description.id);
+    }
+}
+
+#[derive(Message)]
+#[rtype(result = "()")]
+pub struct DeviceUpdated {
+    pub device: Device,
+}
+
+impl Handler<DeviceUpdated> for Device {
+    type Result = ();
+
+    fn handle(&mut self, msg: DeviceUpdated, _ctx: &mut Context<Self>) -> Self::Result {
+        info!("Device updated {}", msg.device.description.id)
+    }
+}
```

Comment:
This seems to be dead code. And also, I am a bit confused why the `DeviceUpdatedMessage` which is received by a device should contain the very same device?